cmd/contour/serve.go

```diff
@@ -187,6 +187,10 @@ func validateCRDs(dynamicClient dynamic.Interface, log logrus.FieldLogger) {
 
 // doServe runs the contour serve subcommand.
 func doServe(log logrus.FieldLogger, ctx *serveContext) error {
+	ectx := context.Background()
+	serveContext, cancel := context.WithCancel(ectx)
```

Comment:
maybe consider renaming `serveContext` since it sorta conflicts with the other `ctx *serveContext` parameter (which is also a bit misleading since I don't think it implements `context.Context`)

Reply:
We could also rename the `serveContext` type and `ctx` param since we have actual `context.Context`s involved here.

Another option would be to make `serveContext` implement `context.Context` but that might be overkill

Reply:
yeah it's unfortunate that it has `context` in the name which does make it confusing.